File: Gymify.Services/Database/Reward.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Gymify.Services.Database
{
    public class Reward
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(150)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Description { get; set; }

    [Required]
    public int RequiredPoints { get; set; }

    public DateTime ValidFrom { get; set; }
    public DateTime ValidTo { get; set; }

    [Required]
    [MaxLength(30)]
    public string Status { get; set; } = "Planned";

    public bool IsLockedForEdit { get; set; }
    public bool CanDelete { get; set; }
    public int RedemptionCount { get; set; }

    public ICollection<UserReward> UserRewards { get; set; } = new List<UserReward>();
}
}

File: UI/gymify_desktop/lib/models/reward.dart
import 'package:json_annotation/json_annotation.dart';

part 'reward.g.dart';

@JsonSerializable()
class Reward {
  final int id;
  final String name;
  final String? description;
  final int requiredPoints;
  final String status;
  final DateTime validFrom;
  final DateTime validTo;
  final bool isLockedForEdit;
  final bool canDelete;
  final int redemptionCount;

  Reward({
    required this.id,
    required this.name,
    this.description,
    required this.requiredPoints,
    required this.status,
    required this.validFrom,
    required this.validTo,
    required this.isLockedForEdit,
    required this.canDelete,
    required this.redemptionCount,
  });

  factory Reward.fromJson(Map<String, dynamic> json) => _$RewardFromJson(json);

  Map<String, dynamic> toJson() => _$RewardToJson(this);
}